libs/GraphicECS/SFML/Systems/InputManagement.cpp

```diff
@@ -165,50 +198,38 @@ namespace graphicECS::SFML::Systems
     {
         std::vector<std::shared_ptr<Entity>> player = world.joinEntities<Controlable>();
 
-        if (player.empty() || action < 1)
+        if (player.empty())
             return;
-        auto shoot = [&world](std::shared_ptr<Entity> entityPtr) {
-            auto entityGuard = std::lock_guard(*entityPtr.get());
-            ShootingFrequency &freq = entityPtr.get()->getComponent<ShootingFrequency>();
-            const char hex_char[] = "0123456789ABCDEF";
-            RandomDevice &random = world.getResource<RandomDevice>();
-            auto guard = std::lock_guard(random);
-            std::string uuid(16, '\0');
-
-            if (freq.frequency == duration<double>(0.0)) {
-                for (auto &c : uuid)
-                    c = hex_char[random.randInt<int>(0, 15)];
-                createNewAlliedProjectile(world, *entityPtr, uuid);
-                freq.frequency = freq.baseFrequency;
+        for (std::shared_ptr<ecs::Entity> p : player) {
+            if (action > 1 && !(p->contains<IsShootingComponent>())) {
```

Comment:
Perhaps you should place the lock in the for scope and not in the if scope because in both cases, you are locking the entity